app/src/main/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt

```diff
@@ -35,22 +82,68 @@ class SearchLocationActivity : AppCompatActivity() {
                 // Do your search
                 searchLocationViewModel.saveSearch(query)
                 val startMainActivity = Intent(this@SearchLocationActivity, MainActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
-
                 if (fromSearchFragment) {
                     val searchBundle = Bundle()
                     searchBundle.putBoolean(TO_SEARCH, true)
                     startMainActivity.putExtras(searchBundle)
                 }
-
                 startActivity(startMainActivity)
-
                 return false
             }
 
             override fun onQueryTextChange(newText: String): Boolean {
                 return false
             }
         })
+
+        current_location.setOnClickListener{
+            progress_bar.visibility = View.VISIBLE
+            val permission = ContextCompat.checkSelfPermission(this,Manifest.permission.ACCESS_COARSE_LOCATION)
+            if (permission != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(this,
+                    arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION),MY_PERMISSIONS_REQUEST)
+            }
```

Comment:
Return if you don't have permission

Reply:
Displaying Toast in onRequestPermissionResult for not giving permission. specific changes other than that for not having permission?

Reply:
Add a return statement in the if condition